src/H5.c

```diff
@@ -949,7 +966,51 @@ H5check_version(unsigned majnum, unsigned minnum, unsigned relnum)
                 break;
         } /* end switch */
 
-    } /* end if */
+    } /* end if (H5_VERS_MAJOR != majnum || H5_VERS_MINOR != minnum) */
+
+    /* H5_VERS_RELEASE should be compatible, we will only add checks for exceptions */
+    if (H5_VERS_RELEASE != relnum) {
+        for (unsigned i = 0; i < VERS_RELEASE_EXCEPTIONS_SIZE; i++) {
+            /* Check for incompatible headers or incompatible library */
+            if (VERS_RELEASE_EXCEPTIONS[i] == relnum || VERS_RELEASE_EXCEPTIONS[i] == H5_VERS_RELEASE) {
```

Comment:
This check depends on the relnum value from the headers not matching the H5_VERS_RELEASE value from the library.  Then we check both against the list of exceptions.